### DISCUSSION TITLE
tunneling network packets

### DISCUSSION THREAD

**Discussion by unicomp21** [jart/blink#22] [General]
Assuming we will get the webassembly stuff working eventually.  If I'm wanting to tunnel network packets via websocket etc. into blink, where should I start in the code?

**Comment by jart** [comment]
I don't understand the question. Could you be more specific? What's your goal?

**Comment by unicomp21** [comment]
To forward network packets in/out of the vm.

**Reply by jart** [reply]
I have no idea. I don't have experience with WASM.

**Reply by unicomp21** [reply]
I don't think it has anything to do w/ wasm?  Is there an example nic (network driver) in the code?

**Reply by pannous** [reply]
If emscripten doesn't provide bindings, you certainly need to do it through asyncify:
https://emscripten.org/docs/porting/asyncify.html
